Library/Homebrew/cmd/reinstall.rb

```diff
@@ -2,6 +2,14 @@
 
 module Homebrew
   def reinstall
+    if !MacOS.can_build?
```

Comment:
In this case, maybe the logic can be moved out of `FormulaInstaller#install` and have the install command call it early in the same way that occurs with reinstall and upgrade?